checks/restic.go

```diff
@@ -45,18 +45,41 @@ func (c *ResticChecker) Check(ctx *context.Context, extConfig external.Check) pk
 	result := pkg.Success(check, ctx.Canary)
 	var results pkg.Results
 	results = append(results, result)
-	envVars, err := c.getEnvVars(check, ctx.Canary.Namespace, ctx.Kommons)
-	if err != nil {
-		return results.Failf("error getting envVars %v", err)
+
+	var envVars map[string]string
+	var err error
+	if check.ConnectionName != "" {
+		connection, err := ctx.HydrateConnectionByURL(check.ConnectionName)
+		if err != nil {
+			return results.Failf("error getting connection: %v", err)
+		}
+
+		envVars = map[string]string{
+			resticPasswordEnvKey: connection.Password,
+			// resticAwsSecretAccessKey:   secretKey,
+			// resticAwsAccessKeyIDEnvKey: connection,
+		}
+
+		// TODO: Not sure why wee need AWS credentials here.
+		// If it's required, will need to think how a single connection can provide
+		// both AWS credentials and a restic password.
```

Comment:
@moshloop

Reply:
Restic stores its backups in AWS S3

Reply:
okay. Password is already taken so I'll use the properties to get the AWS keys.

Reply:
Create new field for aws connection